pages/_app.tsx

```diff
@@ -20,21 +21,40 @@ interface MyAppProps extends AppProps {
 
 export default function MyApp(props: MyAppProps) {
   const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;
-  return (
-    <CacheProvider value={emotionCache}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name='viewport' content='initial-scale=1, width=device-width' />
-      </Head>
-      <ThemeProvider theme={theme}>
-        {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
-        <CssBaseline />
-        <SocketProvider>
-          <UserProvider>
-            <Component {...pageProps} />
-          </UserProvider>
-        </SocketProvider>
-      </ThemeProvider>
-    </CacheProvider>
-  );
+
+  const denyList = [
+    '/teacher/classroom/[classroomName]',
+    '/student/classroom/[classroomName]',
+  ];
+  const router = useRouter();
+  const [userIsAuthorized, setUserIsAuth] = React.useState(false);
```

Comment:
Lets add a comment explanation. I copied/pasted the words in the your PR.
```suggestion
  const [userIsAuthorized, setUserIsAuth] = React.useState(false);
  /* Prevents Unauthorized Page Access to Teacher or Student Pages by checking what page the app starts on in the browser. If the app starts on a page in the denyList then the user is forced back to the homepage. This way all navigation to protected pages must happen from the homepage. */
```